### PR TITLE
Change the return type of `RRuleSet::all` to a struct

### LINKED ISSUE #64: Create release from Git tag with GH actions
As title says.

### LINKED ISSUE #80: Returning `(Vec<DateTime>, bool)` is a bad practice
It's unknown what are the meaning of the elements in the `(Vec<DateTime>, bool)` tuple in the response of `all` method and some others.

Better to return a struct with proper naming and methods.

### PR DESCRIPTION
**Pull request by omid**
Fix #80 
@fmeringdal is “limited” a correct wording? I still don't know its use case :)
Isn't “truncated” better?

Fix #64
I didn't test it, but don't forget to add your API token as `CARGO_REGISTRY_TOKEN` in the repo Settings > Secrets > Actions

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -4,17 +4,21 @@ All notable changes to this project will be documented in this file.
 The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
-## [Unreleased]
+## [Unreleased] (2022-0X-YY)
+
+### Changed
+
+- `RRuleSet::all` returns an `RRuleSetResult` struct instead of a tuple.
```
Comment (fmeringdal) [inline CHANGELOG.md:11]: `RRuleSetResult` seems a little long, maybe simplify it to `RRuleResult`?

File: rrule/src/core/rruleset.rs [hunk 2]
```diff
@@ -31,6 +31,14 @@ pub struct RRuleSet {
     pub(crate) limited: bool,
 }
 
+/// The return result of `RRuleSet::all`.
+pub struct RRuleSetResult {
+    /// List of recurrences.
+    pub list: Vec<DateTime>,
```
Comment (fmeringdal) [inline rrule/src/core/rruleset.rs:37]: How about `dates` rather than `list`?

**Review by fmeringdal — CHANGES_REQUESTED** [review]
Great stuff! Some minor comments, I will make sure to add the cargo secret to the repo secrets.

**Comment by omid** [comment]
Thanks @fmeringdal 
What about my first comment/question?

> @fmeringdal is “limited” a correct wording? I still don't know its use case :)
Isn't “truncated” better?

**Comment by fmeringdal** [comment]
I think limited is a good enough term as it tells that the result is "limited" and could potentially contain more dates and the iteration was "limited" to avoid infinite loops. Open to other suggestions, but I still think limited is more accurate than truncated.

**Comment by omid** [comment]
@fmeringdal cool, np, will add more doc comment soon, and then I'm good I think

**Comment by fmeringdal** [comment]
Looks good! Only need to update the example in the README and it should be good to merge

**Comment by fmeringdal** [comment]
Thanks @omid !

**Pull request merged**